### PR TITLE
Cadd tool

### PR DESCRIPTION
**Pull request by nojux-official**
Changes:
- Implemented functions to fetch SNV scores from CADD online API via single score and range access. 
- Usage section for tools in README.
- Enabled the script to be called with arguments.

### CODE REVIEW

File: tools/cadd/cadd.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            return data
+        print(f"Error: {response.status_code} - {response.text}")
+        return None
+    except requests.exceptions.Timeout:
+        print("Error: Timeout occurred while trying to reach the server.")
+    except requests.exceptions.RequestException as req_err:
+        print(f"Error: {req_err}")
+    except ValueError:
+        print("Error: Invalid JSON format in response.")
+    return None
```
Comment (Strexas) [inline tools/cadd/cadd.py:60]: Raise an exception

File: tools/cadd/cadd.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    if args.position:
+        result = fetch_cadd_score(args.version, args.chromosome, args.position)
+        print(result)
+    elif args.start and args.end:
+        result = fetch_cadd_scores(args.version, args.chromosome, args.start, args.end)
+        print(result)
+    else:
+        print("Please provide either '--position' for single SNV \
+              or '--start' and '--end' for a range of positions.")
```
Comment (Strexas) [inline tools/cadd/cadd.py:81]: In tests you will find pipeline.ipynb where collected data is showed. create new cell and provide example of usage there for whole dataFrame. If not all columns are displayed, search for pandas.options for max_columns

File: tools/cadd/cadd.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        print(f"Error: {response.status_code} - {response.text}")
+        return None
+    except requests.exceptions.Timeout:
+        print("Error: Timeout occurred while trying to reach the server.")
+    except requests.exceptions.RequestException as req_err:
+        print(f"Error: {req_err}")
+    except ValueError:
+        print("Error: Invalid JSON format in response.")
+    return None
+
```
Comment (Strexas) [inline tools/cadd/cadd.py:33]: raise an exception

File: tools/cadd/cadd.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    parser.add_argument("--position", type=int, help="Genomic position (for single SNV)")
+    parser.add_argument("--start", type=int,
+                        help="Genomic start position (for a range of positions)")
+    parser.add_argument("--end", type=int, help="Genomic end position (for a range of positions)")
+
+    args = parser.parse_args()
+
+    if args.position:
+        result = fetch_cadd_score(args.version, args.chromosome, args.position)
+        print(result)
```
Comment (Strexas) [inline tools/cadd/cadd.py:75]: How can I get these results for further processing?

**Review by Strexas — CHANGES_REQUESTED** [review]

File: tools/cadd/cadd.py (hunk 4 above)
Comment (nojux-official) [reply tools/cadd/cadd.py:83]: I've added the usage in pipeline notebook

**Comment by Strexas** [comment]
Closed as not relevant and fully used.

**Pull request closed without merging**